contracts/crosschain/ERC7786Aggregator.sol

```diff
@@ -0,0 +1,278 @@
+    error ERC7786AggregatorThresholdViolation();
+
+    /****************************************************************************************************************
+     *                                        S T A T E   V A R I A B L E S                                         *
+     ****************************************************************************************************************/
+
+    /// @dev address of the matching router for a given CAIP2 chain
+    mapping(string caip2 => string) private _remoteRouters;
+
+    /// @dev Tracking of the received message pending final delivery
+    mapping(bytes32 id => Tracker) private _trackers;
+
+    /// @dev List of authorized IERC7786 gateways (M is the length of this set)
+    EnumerableSet.AddressSet private _gateways;
+
+    /// @dev Threshold for message reception (the threshold opf the sending side is applied on the receiving side)
+    uint8 private _threshold;
+
+    /// @dev Nonce for message deduplication (internal)
+    uint256 private _nonce;
+
+    /****************************************************************************************************************
+     *                                        E V E N T S   &   E R R O R S                                         *
+     ****************************************************************************************************************/
+    event RemoteRegistered(string chainId, string router);
+    error RemoteAlreadyRegistered(string chainId);
+
+    /****************************************************************************************************************
+     *                                              F U N C T I O N S                                               *
+     ****************************************************************************************************************/
+    constructor(address owner_, address[] memory gateways_, uint8 threshold_) Ownable(owner_) {
+        for (uint256 i = 0; i < gateways_.length; ++i) {
+            _addGateway(gateways_[i]);
+        }
+        _setThreshold(threshold_);
+    }
+
+    // ============================================ IERC7786GatewaySource ============================================
+
+    /// @inheritdoc IERC7786GatewaySource
+    function supportsAttribute(bytes4 selector) external view returns (bool) {
+        for (uint256 i = 0; i < _gateways.length(); ++i)
+            if (!IERC7786GatewaySource(_gateways.at(i)).supportsAttribute(selector)) return false;
+        return true;
+    }
+
+    /// @inheritdoc IERC7786GatewaySource
+    function sendMessage(
+        string calldata destinationChain,
+        string memory receiver, // using memory instead of calldata avoids stack too deep error
+        bytes memory payload, // using memory instead of calldata avoids stack too deep error
+        bytes[] memory attributes // using memory instead of calldata avoids stack too deep error
```

Comment:
```suggestion
    /// @inheritdoc IERC7786GatewaySource
    /// @dev Using memory instead of calldata avoids stack too deep errors
    function sendMessage(
        string calldata destinationChain,
        string memory receiver,
        bytes memory payload,
        bytes[] memory attributes
```